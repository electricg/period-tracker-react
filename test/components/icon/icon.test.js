import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../../../src/components/icon';

test('should fail returning nothing if no id is passed', () => {
  const component = renderer.create(
    <Icon className="someclassname" cssModifier="somecssmodifier" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should succeed when no cssModifier is passed', () => {
  const component = renderer.create(
    <Icon id="someid" className="someclassname" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should succeed when no className is passed', () => {
  const component = renderer.create(
    <Icon id="someid" cssModifier="somecssmodifier" />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
