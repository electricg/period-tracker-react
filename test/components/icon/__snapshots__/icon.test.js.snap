// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fail returning nothing if no id is passed 1`] = `null`;

exports[`should succeed when no className is passed 1`] = `
<svg
  className="icon icon--somecssmodifier"
  role="presentation"
>
  <use
    xlinkHref="#icon-someid"
  />
</svg>
`;

exports[`should succeed when no cssModifier is passed 1`] = `
<svg
  className="icon someclassname"
  role="presentation"
>
  <use
    xlinkHref="#icon-someid"
  />
</svg>
`;
